frontend/src/pages/NewRunV2.tsx

```diff
@@ -246,31 +311,58 @@ function NewRunV2(props: NewRunV2Props) {
       <div className={commonCss.scrollContainer}>
         <div className={commonCss.header}>Run details</div>
 
+        {!pipelineId && apiResourceRefFromRun && !apiResourceRefFromRun[1] && (
+          <div>
+            <div>
+              <span>{usePipelineFromRunLabel}</span>
+            </div>
+            <div className={classes(padding(10, 't'))}>
+              {originalRunId && (
+                <Link className={classes(commonCss.link)} to={pipelineDetailsUrl}>
+                  [View pipeline]
+                </Link>
+              )}
+            </div>
+          </div>
+        )}
+
         {/* Pipeline selection */}
-        <Input
-          value={apiPipeline?.name || ''}
-          required={true}
-          label='Pipeline'
-          disabled={true}
-          variant='outlined'
-          InputProps={{
-            classes: { disabled: css.nonEditableInput },
-            readOnly: true,
-          }}
-        />
+        {(pipelineId || (apiResourceRefFromRun && apiResourceRefFromRun[1])) && (
+          <Input
+            value={
+              (apiResourceRefFromRun !== undefined
+                ? apiResourceRefFromRun[1].name
```

Comment:
Currently use pipeline version name as pipeline name for cloning run.

Reply:
Maybe I am missing context: What does `apiResourceRefFromRun[1]` mean here? Does that mean pipeline version name? Looks like we are looking for the second element in resource reference array, which doesn't seem to be a reliable condition to check.

On the other hand, if this is actually the case, it worths moving your explanation here to the code itself as comments.

Reply:
Yes, I am trying to get the pipeline version name from RunDetails. Not sure if there's any other reference can provide apiVersion name.

Reply:
For `Clone Run`, we only need to retrieve IR from ApiRun object. And we don't expect pipeline and pipelineversion in this case.